tidy3d/plugins/smatrix/smatrix.py

```diff
@@ -92,31 +95,72 @@ class ComponentModeler(Tidy3dBaseModel):
         title="Folder Name",
         description="Name of the folder for the tasks on web.",
     )
-
-    batch: Tidy3dBaseModel = pd.Field(
+    element_mappings: Dict[Element, Tuple[Element, bool]] = pd.Field(
```

Comment:
Yeah this is general enough alright but certainly not super convenient. This mapping can also be constructed by an isomorphism relationship like (p0, p1, p2) (p0, p2, p1) and an `invert_sign` list of `(Port, mode_index)` corresponding to that particular isomorphism. E.g. in the splitter case, the list will contain all the modes of port 0 that are antisymmetric. Maybe we can have a function that takes input like that and returns an `element_mappings` dict to be fed in here?

Reply:
I'm not sure exactly how the `invert_sign` is handled. So if `invert_sign = [p1, 0]` does that mean that any element mappings from `p0, 0` to *any* other port should have an inverted sign? and everything else should not have an inverted sign? otherwise, I agree that an alternative constructor like this would be useful.

Reply:
> I'm not sure exactly how the `invert_sign` is handled. So if `invert_sign = [(p0, 0)]` does that mean that any element mappings from `(p0, 0)` to _any_ other port should have an inverted sign? and everything else should not have an inverted sign?

I think it needs to be applied whenever the isomorphism enforces something like `S_ij = S_iq` or `S_ji = S_qi`, where `i` is the mode that has "invert_sign" under that isomorphism.

I'm fine with putting this off for now though.